api/lib/infrastructure/events/EventBus.js

```diff
@@ -0,0 +1,52 @@
+const bluebird = require('bluebird');
+
+class SubscriberList {
+  constructor() {
+    this._subscribers = new Map();
+  }
+
+  add(eventClass, subscriber) {
+    if (!this._subscribers.has(eventClass)) {
+      this._subscribers.set(eventClass, []);
+    }
+    this._subscribers.get(eventClass).push(subscriber);
+  }
+
+  get(event) {
+    const subscribersToCall = [];
+
+    this._subscribers.forEach((subscribers, eventClass) => {
+      if (event instanceof eventClass) {
+        subscribersToCall.push(...subscribers);
+      }
+    });
+
+    return subscribersToCall;
+  }
+}
+
+class EventBus {
+  constructor(dependenciesBuilder, errorHandler) {
+    this._subscriptions = new SubscriberList();
+    this._dependenciesBuilder = dependenciesBuilder;
+    this._errorHandler = errorHandler;
+  }
+
+  subscribe(event, subscriber) {
+    this._subscriptions.add(event, subscriber);
+  }
+
+  async publish(event, domainTransaction) {
+    const subscribersList = this._subscriptions.get(event);
+    await bluebird.mapSeries(subscribersList, async (subscriberClass) => {
+      const subscriber = this._dependenciesBuilder.build(subscriberClass, domainTransaction);
```

Comment:
Est-ce que ce ne serait pas plutôt uniquement les subscriber à appeler ici ? (subscriberList laisse sous entendre qu'il y a encore tous les subscriber)
```suggestion
    const subscribersToCall = this._subscriptions.get(event);
    await bluebird.mapSeries(subscribersToCall, async (subscriberClass) => {
```

Reply:
done